### PR TITLE
TST: Adds TestingSlippage slippage model

### PR DESCRIPTION
**Pull request by yankees714**
Allows specifying a constant number of shares filled per tick.

### CODE REVIEW

File: zipline/testing/slippage.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    ----------
+    filled_per_tick : int or TestingSlippage.ALL
+        The number of shares to fill on each call to process_order. If
+        TestingSlippage.ALL is passed, the entire order is filled.
+
+    See also
+    --------
+    zipline.finance.slippage.SlippageModel
+    """
+    ALL = sentinel('All')
```
Comment (ssanderson) [inline zipline/testing/slippage.py]: In most other places, we use the same name for the sentinel value and the string we pass for its repr.  Any reason not to do that here as well?

File: zipline/testing/slippage.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    def __init__(self, filled_per_tick):
+        self.filled_per_tick = filled_per_tick
+
+    def process_order(self, data, order):
+        price = data.current(order.asset, "close")
+        if self.filled_per_tick == self.ALL:
```
Comment (ssanderson) [inline zipline/testing/slippage.py]: For `sentinel` instances `is` is the preferred check.

Comment (yankees714) [inline zipline/testing/slippage.py]: Makes sense 👍

File: zipline/testing/slippage.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+from zipline.finance.slippage import SlippageModel
+from zipline.utils.sentinel import sentinel
+
+
+class TestingSlippage(SlippageModel):
```
Comment (ssanderson) [inline zipline/testing/slippage.py]: We should probably add a test that exercises this if it's going to live in zipline.

**Comment by ssanderson** [comment]
lgtm

**Pull request merged**